databases/week3/meal_sharing.sql

```diff
@@ -0,0 +1,93 @@
+-- Meal Sharing Queries
+
+-- Meal Queries
+-- Get all meals
+SELECT * FROM meal;
+
+-- Add a new meal
+INSERT INTO meal (title, description, location, max_reservations, price, created_date)
+VALUES ("NewMeal", "Some description", "Denmark", 10, 400.00, "2021-08-12");
+
+-- Get a meal with any id, fx 1
+SELECT * FROM meal WHERE meal.id = 1;
+
+-- Update a meal with any id, fx 1.
+-- Update any attribute fx the title or multiple attributes
+UPDATE meal SET price = 250 WHERE meal.id = 1;
+
+-- Delete a meal with any id, fx 1
+DELETE FROM meal WHERE id = 7;
+
+
+-- Reservation Queries
+-- Get all reservations
+SELECT * FROM reservation;
+
+-- Add a new reservation
+INSERT INTO reservation (number_of_guests, meal_id, created_date, contact_phonenumber, contact_name, contact_email)
+VALUES (6, 1, "2021-08-20 17:00:00", "20 10 11 88", "NewContact", "new_contact@email.com");
+
+-- Get a reservation with any id, fx 1
+SELECT * FROM reservation WHERE reservation.id = 1;
+
+-- Update a reservation with any id, fx 1.
+-- Update any attribute fx the title or multiple attributes
+UPDATE reservation SET meal_id = 2 WHERE reservation.id = 1;
+
+-- Delete a reservation with any id, fx 1
+DELETE FROM reservation WHERE id = 7;
+
+
+-- Review Queries
+-- Get all reviews
+SELECT * FROM review;
+
+-- Add a new review
+INSERT INTO review (title, description, meal_id, stars, created_date)
+VALUES ("NewReview", "Some details about the meal", 1, "8", "2021-08-12");
+
+-- Get a review with any id, fx 1
+SELECT * FROM review WHERE review.id = 1;
+
+-- Update a review with any id, fx 1.
+-- Update any attribute fx the title or multiple attributes
+UPDATE review SET meal_id = 4 WHERE review.id = 1;
+
+-- Delete a review with any id, fx 1
+DELETE FROM review WHERE id = 7;
+
+
+-- Additional Queries
+-- Get meals that has a price smaller than a specific price fx 90
+SELECT * FROM meal WHERE meal.price < 400;
+
+-- Get meals that still has available reservations
+SELECT meal.title AS "Title", meal.max_reservations AS "Max Reservations", reservation.number_of_guests AS "Guests Number"
+FROM meal
+JOIN reservation ON meal.id = reservation.meal_id
+WHERE meal.max_reservations > reservation.number_of_guests;
```

Comment:
When meal probably as more than one reservation, and each reservation probably has a different number of guests. You need to sum the guests from all reservations and check if that value is greater than the max_reservations. Does it make sense?
There are different ways of doing it: one with a query inside a query, another using COALESCE: https://dev.mysql.com/doc/refman/8.0/en/comparison-operators.html#function_coalesce